File: Design-Patterns-Using-JavaScript/Adapter/AdapterPattern.js
// Desired Interface
class DesiredInterface {
    operation() {
        throw new Error("Abstract method: operation");
    }
}

// Adapter class
class Adapter extends DesiredInterface {
    constructor() {
        super();
        this.adaptee = new Adaptee();
    }

    operation() {
        this.adaptee.someOperation();
    }
}

// Adaptee class
class Adaptee {
    someOperation() {
        console.log("Adaptee someOperation() function called.");
    }
}

// Client Code
const adapter = new Adapter();
adapter.operation();
/*
Adaptee someOperation() function called.
*/

File: Design-Patterns-Using-Swift/Adapter/AdapterPattern.swift
// Desired Interface
protocol DesiredInterface {
    func operation()
}

// Adapter class
class Adapter: DesiredInterface {
    private var adaptee: Adaptee
    
    init() {
        self.adaptee = Adaptee()
    }
    
    func operation() {
        adaptee.someOperation()
    }
}

// Adaptee class
class Adaptee {
    func someOperation() {
        print("Adaptee someOperation() function called.")
    }
}

// Client Code
let adapter: DesiredInterface = Adapter()
adapter.operation()

/*
Adaptee someOperation() function called.
*/
